### PR TITLE
Install 3rd-party JVM dependencies via `sider.yml`

### LINKED ISSUE #1000: Support third-party dependencies for JVM tools
According to the [document](https://checkstyle.sourceforge.io/config.html), Checkstyle can get written arbitrary modules in its configuration.

Some of our customers want its feature, but I'm not sure how to implement it.

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

This change allows users to install 3rd-party JVM dependencies via `sider.yml`.

For example, users can do it to use 3rd-party Checkstyle rules:

```yaml
linter:
  checkstyle:
    jvm_deps:
      # [group, name, version] format
      - [com.github.sevntu-checkstyle, sevntu-checks, 1.37.1]
```

NOTE:
I suggested `maven` instead of `jvm_deps` in [this comment](https://github.com/sider/runners/issues/1000#issuecomment-657994815), but I think it not good that naming depends on a specific tool name (e.g. Maven, Gradle, etc.).
So, I renamed it to `jvm_deps` and adopted only the format `[group, name, version]`.

> Link related issues or pull requests, e.g. `Fix #123`, `Related to #456`, or `None`.

Fix #1000

> Check the following items (please remove needless items).

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@yykamei This pull request is still in progress, but can you please give me feedback if you have?

File: lib/runners/schema/config.rb [hunk 1]
```diff
@@ -37,6 +37,10 @@ module Schema
       let :npm, (base.update_fields { |fields|
         fields.merge!(npm_install: optional(npm_install))
       })
+
+      let :java, (base.update_fields { |fields|
+        fields.merge!(jvm_deps: array?(array(string)))
```
Comment (yykamei) [inline lib/runners/schema/config.rb:42]: `version` may be `number`?: `[com.foo, bar, 1.2]`.

Comment (ybiquitous) [reply lib/runners/schema/config.rb:42]: Good catch! If we could support `number`, users would feel nice. 👍

**Comment by ybiquitous** [comment]
> Perhaps, we may have to let users add other repositories like JCenter, but it might be rare cases.

Good point. I think this case might occur in the future. It seems good that we consider it now.

Here are my suggestions:

1. Increase repositories by default defined in <https://docs.gradle.org/current/userguide/declaring_repositories.html>:

- `mavenCentral`
- `jcenter`
- `google`

2. Custom repositories, especially in Enterprise, might is necessary; e.g. a company private repository (see also https://cloudsmith.com/maven-repository/)

I'm not sure how to realize it now, but for example:

```yaml
linter:
  checkstyle:
    jvm_repos: # default to [mavenCentral, jcenter, google]
      - foo
      - bar
      - ...
```

In this pull request, I will implement the **1** option, but the **2** option is not necessary for now.
What do you think?

File: lib/runners/schema/config.rb (hunk 1 above)
Comment (ybiquitous) [reply lib/runners/schema/config.rb]: Fixed via 122f02d

**Comment by yykamei** [comment]
> In this pull request, I will implement the **1** option, but the **2** option is not necessary for now.
> What do you think?

I agree. If the option **2** is supplementally implementable later, this PR is enough to implement the option **1**.

**Comment by ybiquitous** [comment]
Thanks. I implemented **1** via cee2aa27.

**Comment by ybiquitous** [comment]
The ktlint processor has a special schema:

https://github.com/sider/runners/blob/7d156cc9d1ad645f96613b3c82aa427935cdea59/lib/runners/processor/ktlint.rb#L28-L31

So, I think it proper that The ktlint processor does not support this feature yet (we may need to reconsider the schema, but breaking).

**Comment by ybiquitous** [comment]
@yykamei Could you please re-review?

**Comment by ybiquitous** [comment]
Thanks for the review. When the CI passes, I will merge this PR. 👌

**Pull request merged**